tests/utils.go

```diff
@@ -2302,6 +2302,13 @@ func NewRandomFedoraVMIWithVirtWhatCpuidHelper() *v1.VirtualMachineInstance {
 	return vmi
 }
 
+func GetFedoraStandardUserData() string {
+	return `#!/bin/bash
+            echo "fedora" |passwd fedora --stdin
```

Comment:
wouldn't it be better to do those two commands (or at least the first)
when building the VM here https://github.com/kubevirt/kubevirtci/pull/530/files#diff-60e708f2e40cf0f5d49420a079fdd43d2a2b2691cada7e745ee1033d195d8a80R5 ?

otherwise the `LoginToFedora` might race with cloud-init setting the password 
causing LoginToFedora required to be robust to corner cases and so on 

seen on WIP https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/851/rehearsal-pull-kubevirt-e2e-kind-1.17-sriov/1351794767805026304

We can see here that there was a login attempt, before the clout-init configured the password on the last lines. 
```
09:27:40: testvmi-g7bcg login: 
09:27:40: [34mSent:[39m "fedora\n"
09:27:40: [32mMatch for RE:[39m "Password:" found: ["Password:"] Buffer: fedora

09:27:40: Password: 
09:27:40: [34mSent:[39m "fedora\n"
09:27:40: [32mMatch for RE:[39m "Login incorrect" found: ["Login incorrect"] Buffer: 
09:27:40: [  111.375055] cloud-init[758]: Cloud-init v. 19.4 running 'modules:config' at Wed, 20 Jan 2021 09:26:12 +0000. Up 110.14 seconds.
09:27:40: Login incorrect
09:27:40: [32mMatch for RE:[39m "(localhost|testvmi-g7bcg) login: " found: ["testvmi-g7bcg login: " "testvmi-g7bcg"] Buffer: 
09:27:40: 
09:27:40: testvmi-g7bcg login: 
09:27:40: [34mSent:[39m "fedora\n"
09:27:40: [32mMatch for RE:[39m "Password:" found: ["Password:"] Buffer: fedora
09:27:40: Password:
09:27:40: [34mSent:[39m "fedora\n"
09:27:40: [32mMatch for RE:[39m "Login incorrect" found: ["Login incorrect"] Buffer:  
09:27:40: [  113.778406] cloud-init[777]: Changing password for user fedora.
09:27:40: [  113.781852] cloud-init[777]: passwd: all authentication tokens updated successfully.
```

more problems about `LoginToFedora` seen in the past on issues that reopen here and there and not resolved yet.

Reply:
interesting. you're right. i am seeing instances where we fight this race condition.

Reply:
I took a slightly different route here. We need a way to determine when cloud-init has been processed and only act on the VM guest once that has occurred. The guest agent has been used by many of the functional tests to perform this task.

By enabling the guest agent in the cloud-init user-data, we can wait on the agent connected condition on the VMI. Once that condition is met, we know coud-init has finished. Then we can log in successfully to the vmi.

Rather that giving the fedora tools vm a default password (which conflicts with some tests), i just made it the agent is disabled at startup and must be started via cloud init. That gives us a consistent way order the login attempt... 

so
- wait on agent connect condition
- then attempt to login

Reply:
cool, so we will need to change order in tests right ? (in case we use the updated VM / same route)
currently we are using this
first wait vm ready + login and then just wait for agent connected
```
			tests.WaitUntilVMIReady(vmi, libnet.WithIPv6(console.LoginToFedora))
			tests.WaitAgentConnected(virtClient, vmi)
```